src/errors/parser.ts

```diff
@@ -8,35 +8,35 @@ import { PathParameterError } from "./PathParameterError.js";
 import { RequestParameterError } from "./RequestParameterError.js";
 import { RequestError, RequestResponseBaseError, ResponseError } from "./RequestResponseError.js";
 
-export const getProcessingErrorCodeByStatus = (status: number): string => {
+export const getProcessingErrorCodeByStatus = (status: number, isThreeDsError?: boolean, message?: string): string => {
     const codeMap: Record<number, string> = {
-        301: ResponseErrorCode.CardNumberNotValid,
+        301: isThreeDsError ? ResponseErrorCode.ThreeDsDefaultError : ResponseErrorCode.CardNumberNotValid,
         302: ResponseErrorCode.CardExpirationMonthInvalid,
         303: ResponseErrorCode.CardExpirationYearInvalid,
-        304: ResponseErrorCode.CardExpired,
+        304: isThreeDsError ? ResponseErrorCode.ThreeDsDefaultError : ResponseErrorCode.CardExpired,
         305: ResponseErrorCode.CardCVVInvalid,
         306: ResponseErrorCode.CardRejected,
-        307: ResponseErrorCode.CardInvalid,
+        307: isThreeDsError ? ResponseErrorCode.ThreeDsDefaultError : ResponseErrorCode.CardInvalid,
         308: ResponseErrorCode.ChargeInvalidData,
-        309: ResponseErrorCode.ProcessingError,
+        309: isThreeDsError ? ResponseErrorCode.ThreeDsDefaultError : ResponseErrorCode[message],
```

Comment:
Please display the processing error when the message is not found in the error code to keep the current handling.

Also, please add the missing messages in the `ResponseErrorCode` if any 🙇